warp/sim/import_mjcf.py

```diff
@@ -361,125 +624,129 @@ def parse_body(body, parent, incoming_defaults: dict):
-                        half_height=geom_height,
-                        density=density,
-                        up_axis=geom_up_axis,
-                        **contact_vars,
-                    )
+                    visuals.append(geom)
 
+        if parse_visuals_as_colliders:
+            colliders = visuals
+        else:
+            s = parse_shapes(
+                defaults, body_name, link, visuals, density=0.0, just_visual=True, visible=not hide_visuals
+            )
+            visual_shapes.extend(s)
+
+        show_colliders = force_show_colliders
+        if parse_visuals_as_colliders:
+            show_colliders = True
+        elif len(visuals) == 0:
+            # we need to show the collision shapes since there are no visual shapes
+            show_colliders = True
+
+        parse_shapes(defaults, body_name, link, colliders, density, visible=show_colliders)
+
+        m = builder.body_mass[link]
+        if not ignore_inertial_definitions and body.find("inertial") is not None:
+            inertial = body.find("inertial")
+            if "inertial" in defaults:
+                inertial_attrib = merge_attrib(defaults["inertial"], inertial.attrib)
             else:
-                print(f"MJCF parsing shape {geom_name} issue: geom type {geom_type} is unsupported")
+                inertial_attrib = inertial.attrib
+            # overwrite inertial parameters if defined
+            inertial_pos = parse_vec(inertial_attrib, "pos", (0.0, 0.0, 0.0)) * scale
+            inertial_rot = parse_orientation(inertial_attrib)
+
+            inertial_frame = wp.transform(inertial_pos, inertial_rot)
+            com = inertial_frame.p
+            if inertial_attrib.get("diaginertia") is not None:
+                diaginertia = parse_vec(inertial_attrib, "diaginertia", None)
+                I_m = np.zeros((3, 3))
+                I_m[0, 0] = diaginertia[0] * scale**2
+                I_m[1, 1] = diaginertia[1] * scale**2
+                I_m[2, 2] = diaginertia[2] * scale**2
+            else:
+                fullinertia = inertial_attrib.get("fullinertia")
+                assert fullinertia is not None
+                fullinertia = np.fromstring(fullinertia, sep=" ", dtype=np.float32)
+                I_m = np.zeros((3, 3))
+                I_m[0, 0] = fullinertia[0] * scale**2
+                I_m[1, 1] = fullinertia[1] * scale**2
+                I_m[2, 2] = fullinertia[2] * scale**2
+                I_m[0, 1] = fullinertia[3] * scale**2
+                I_m[0, 2] = fullinertia[4] * scale**2
+                I_m[1, 2] = fullinertia[5] * scale**2
+                I_m[1, 0] = I_m[0, 1]
+                I_m[2, 0] = I_m[0, 2]
+                I_m[2, 1] = I_m[1, 2]
+            rot = wp.quat_to_matrix(inertial_frame.q)
+            I_m = rot @ wp.mat33(I_m)
+            m = float(inertial_attrib.get("mass", "0"))
+            builder.body_mass[link] = m
+            builder.body_inv_mass[link] = 1.0 / m
```

Comment:
Can you check that the `body_inv_mass` and `body_inv_inertia` are set to finite values in case of zero masses/inertia? We have this in the `ModelBuilder.add_body` method for example:
```py
if m > 0.0:
    self.body_inv_mass.append(1.0 / m)
else:
    self.body_inv_mass.append(0.0)

if any(x for x in inertia):
    self.body_inv_inertia.append(wp.inverse(inertia))
else:
    self.body_inv_inertia.append(inertia)
```

Reply:
Done! https://github.com/NVIDIA/warp/pull/538/commits/a8a4d858c43d3a1dbb3a7ea0b9cb19557e6b9784